packages/client/lib/sync/fetcher/accountfetcher.ts

```diff
@@ -0,0 +1,329 @@
+      ? bigIntToBuffer(bufferToBigInt(partialResult[partialResult.length - 1].hash) + BigInt(1))
+      : bigIntToBuffer(first)
+    return setLengthLeft(origin, 32)
+  }
+
+  private getLimit(job: Job<JobTask, AccountData[], AccountData>): Buffer {
+    const { task } = job
+    const { first, count } = task
+    const limit = bigIntToBuffer(first + BigInt(count) - BigInt(1))
+    return setLengthLeft(limit, 32)
+  }
+
+  /**
+   * Request results from peer for the given job.
+   * Resolves with the raw result
+   * If `undefined` is returned, re-queue the job.
+   * @param job
+   * @param peer
+   */
+  async request(
+    job: Job<JobTask, AccountData[], AccountData>
+  ): Promise<AccountDataResponse | undefined> {
+    const { peer } = job
+    const origin = this.getOrigin(job)
+    const limit = this.getLimit(job)
+
+    const rangeResult = await peer!.snap!.getAccountRange({
+      root: this.root,
+      origin,
+      limit,
+      bytes: BigInt(this.config.maxRangeBytes),
+    })
+
+    const peerInfo = `id=${peer?.id.slice(0, 8)} address=${peer?.address}`
+
+    // eslint-disable-next-line eqeqeq
+    if (rangeResult === undefined) {
+      return undefined
+    } else {
+      // validate the proof
+      try {
+        // verifyRangeProof will also verify validate there are no missed states between origin and
+        // response data
+        await this.verifyRangeProof(this.root, origin, rangeResult)
+
+        // Check if there is any pending data to be synced to the right
+        let completed: boolean
+        if (this.isMissingRightRange(limit, rangeResult)) {
+          this.debug(
+            `Peer ${peerInfo} returned missing right range account=${rangeResult.accounts[
+              rangeResult.accounts.length - 1
+            ].hash.toString('hex')} limit=${limit.toString('hex')}`
+          )
+          completed = false
+        } else {
+          completed = true
+        }
+        return Object.assign([], rangeResult.accounts, { completed })
+      } catch (err) {
+        throw Error(`InvalidAccountRange: ${err}`)
+      }
+    }
+  }
+
+  /**
+   * Process the reply for the given job.
+   * If the reply contains unexpected data, return `undefined`,
+   * this re-queues the job.
+   * @param job fetch job
+   * @param result result data
+   */
+  process(
+    job: Job<JobTask, AccountData[], AccountData>,
+    result: AccountDataResponse
+  ): AccountData[] | undefined {
+    const fullResult = (job.partialResult ?? []).concat(result)
+    job.partialResult = undefined
+    if (result.completed === true) {
+      return fullResult
+    } else {
+      // Save partial result to re-request missing items.
+      job.partialResult = fullResult
+    }
+  }
+
+  /**
+   * Store fetch result. Resolves once store operation is complete.
+   * @param result fetch result
+   */
+  async store(result: AccountData[]): Promise<void> {
+    this.debug(`Stored ${result.length} accounts in account trie`)
```

Comment:
right now store doesn't do anything, as a separate helper class with be added in a followup PR to track and aid the fetcher